rct229/rulesets/ashrae9012019/section5/section5rule1.py

```diff
@@ -0,0 +1,285 @@
+                "has_baseline_270": has_baseline_270,
+                "has_proposed": has_proposed,
+                "has_user": has_user,
+                "has_basseline_0_output": has_basseline_0_output,
+                "has_basseline_90_output": has_basseline_90_output,
+                "has_basseline_180_output": has_basseline_180_output,
+                "has_basseline_270_output": has_basseline_270_output,
+                "has_proposed_output": has_proposed_output,
+                "has_user_output": has_user_output,
+                "no_of_rmds": no_of_rmds,
+                "no_of_output_instance": no_of_output_instance,
+            }
+
+        class BuildingRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section5Rule1.RMDRule.BuildingRule, self).__init__(
+                    rmrs_used=produce_ruleset_model_instance(
+                        USER=False,
+                        BASELINE_0=True,
+                        BASELINE_90=False,
+                        BASELINE_180=False,
+                        BASELINE_270=False,
+                        PROPOSED=False,
+                    ),
+                    required_fields={
+                        "$.building_segments[*].zones[*].surfaces[*]": ["azimuth"],
+                    },
+                    fail_msg="Fail unless Table G3.1#5a exception #2 is applicable and it can be demonstrated that the building orientation is dictated by site considerations.",
+                )
+
+            def get_calc_vals(self, context, data=None):
+                building_b0 = context.BASELINE_0
+
+                has_baseline_0 = data["has_baseline_0"]
+                has_baseline_90 = data["has_baseline_90"]
+                has_baseline_180 = data["has_baseline_180"]
+                has_baseline_270 = data["has_baseline_270"]
+                has_proposed = data["has_proposed"]
+                has_user = data["has_user"]
+                has_basseline_0_output = data["has_basseline_0_output"]
+                has_basseline_90_output = data["has_basseline_90_output"]
+                has_basseline_180_output = data["has_basseline_180_output"]
+                has_basseline_270_output = data["has_basseline_270_output"]
+                has_proposed_output = data["has_proposed_output"]
+                has_user_output = data["has_user_output"]
+                no_of_rmds = data["no_of_rmds"]
+                no_of_output_instance = data["no_of_output_instance"]
+
+                # define a function to get the azimuth's corresponding key
+                def get_key_for_azi(azi):
+                    azi_value = azi.to("degrees").m
+                    low_bound = azi_value - (azi_value % 3)
+                    high_bound = low_bound + 3
+                    return f"{low_bound}-{high_bound}"
+
+                azimuth_fen_area_dict_b = {}
+                total_surface_fenestration_area_b = ZERO.AREA
```

Comment:
` total_surface_fenestration_area_b` needs to be reset to 0 in each iteration of surface

Reply:
Addressed!